AccessCheckoutSDK/AccessCheckoutSDKTests/validation/card/AccessCardValidationStateHandlerTests.swift

```diff
@@ -0,0 +1,185 @@
+        XCTAssertFalse(validationStateHandler.panValidationState)
+
+        validationStateHandler.handle(result: (isValid: true, cardBrand: nil))
+
+        XCTAssertTrue(accessCardDelegate.panValidationCalled)
+        XCTAssertTrue(validationStateHandler.panValidationState)
+
+    }
+    
+    func testShouldNotifyMerchantDelegateWithCardBrandIfCardBrandChanges() {
+        let accessCardDelegate = AccessCardDelegateMock()
+
+        let validationStateHandler = AccessCardValidationStateHandler(
+            accessCardDelegate: accessCardDelegate
+        )
+        
+        XCTAssertNil(validationStateHandler.cardBrand)
+        XCTAssertFalse(accessCardDelegate.cardBrandCalled)
+        
+        validationStateHandler.handle(result: (isValid: false, cardBrand: visaBrand))
+
+        XCTAssertEqual(validationStateHandler.cardBrand?.name, "visa")
+        XCTAssertEqual(accessCardDelegate.cardBrand?.name, "visa")
+        XCTAssertTrue(accessCardDelegate.cardBrandCalled)
+
+    }
+    
+    func testShouldNotNotifyMerchantDelegateIfCardBrandDoesNotChange() {
+        let accessCardDelegate = AccessCardDelegateMock()
+
+        let validationStateHandler = AccessCardValidationStateHandler(
+            accessCardDelegate: accessCardDelegate,
+            panValidationState: false,
+            cardBrand: visaBrand
+        )
+                
+        XCTAssertEqual(validationStateHandler.cardBrand?.name, "visa")
+        XCTAssertFalse(accessCardDelegate.cardBrandCalled)
+        
+        validationStateHandler.handle(result: (isValid: false, cardBrand: visaBrand))
+
+        XCTAssertEqual(validationStateHandler.cardBrand?.name, "visa")
+        XCTAssertFalse(accessCardDelegate.cardBrandCalled)
+    }
+    
+    func testShouldReturnFalseIfCardBrandDidNotChange() {
+        let validationStateHandler = AccessCardValidationStateHandler(
+            accessCardDelegate: accessCardDelegate,
+            panValidationState: false,
+            cardBrand: visaBrand
+        )
+                
+        XCTAssertFalse(validationStateHandler.cardBrandChanged(cardBrand: visaBrand))
+    }
+    
+    func testShouldReturnTrueIfCardBrandDidChangeFromNil() {
+        let validationStateHandler = AccessCardValidationStateHandler(
+            accessCardDelegate: accessCardDelegate
+        )
+        
+        XCTAssertNil(validationStateHandler.cardBrand)
+                
+        XCTAssertTrue(validationStateHandler.cardBrandChanged(cardBrand: visaBrand))
+    }
+    
+    func testShouldReturnTrueIfCardBrandDidChangeFromAnotherBrand() {
+        let validationStateHandler = AccessCardValidationStateHandler(
+            accessCardDelegate: accessCardDelegate,
+            panValidationState: false,
+            cardBrand: visaBrand
+        )
+        
+        XCTAssertTrue(validationStateHandler.cardBrandChanged(cardBrand: amexBrand))
+    }
+    
+    func testShouldReturnTrueIfCardBrandDidChangeFromABrandToNil() {
```

Comment:
Rename these tests so that they reflect we just test whther the card brand in parameter is the same as the one held in state